python/python/tests/test_util.py

```diff
@@ -84,3 +85,37 @@ def test_local_join_uri_windows():
         assert joined == str(pathlib.Path(base) / "table.lance")
         joined = join_uri(pathlib.Path(base), "table.lance")
         assert joined == pathlib.Path(base) / "table.lance"
+
+
+def test_value_to_sql_string(tmp_path):
+    # Make sure we can convert Python string literals to SQL strings, even if
+    # they contain characters meaningful in SQL, such as ' and \.
+    values = [
+        "anthony's",
+        "test\n newline",
+        "anthony's \n newline",
+    ]
+    expected_values = [
+        '"anthony\'s"',
+        '"test\n newline"',
+        '"anthony\'s \n newline"',
+    ]
+
+    for value, expected in zip(values, expected_values):
+        assert value_to_sql(value) == expected
+
+    # Also test we can roundtrip those strings through update.
+    # This validates the query parser understands the strings we
+    # are creating.
+    db = lancedb.connect(tmp_path)
+    table = db.create_table(
+        "test",
+        [
+            {"search": "anthony's", "replace": "something"},
+            {"search": "test\n newline", "replace": "something"},
+            {"search": "anthony's \n newline", "replace": "something"},
+        ],
```

Comment:
Seems like we shouldn't need to repeat those values:
```suggestion
        [{"search": value, "replace": "something"} for value in values],
```

Reply:
Sure. I went for clarity. I will update it.